test/clj/rems/db/test_csv.clj

```diff
@@ -0,0 +1,112 @@
+(ns ^:integration rems.db.test-csv
+  (:require [clojure.test :refer :all]
+            [rems.db.applications :as applications]
+            [rems.db.test-data :as test-data]
+            [rems.db.testing :refer [rollback-db-fixture test-db-fixture]]
+            [rems.db.csv :as csv]
+            [rems.text :as text])
+  (:import [org.joda.time DateTime]))
+
+(use-fixtures :once test-db-fixture)
+(use-fixtures :each rollback-db-fixture)
+
+(deftest test-print-to-csv
+  (testing "with default settings"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["val1" "val2"]
+                                    ["val3" "val4"]]
+                             :separator ",")
+           "col1,col2\nval1,val2\nval3,val4\n")))
+
+  (testing "encloses strings in quotes"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["val1" "val2"]
+                                    ["val3" "val4"]]
+                             :separator ","
+                             :enclose-strings-in-quotes true)
+           "\"col1\",\"col2\"\n\"val1\",\"val2\"\n\"val3\",\"val4\"\n")))
+
+  (testing "does not enclose numeric values in quotes"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["val1" "val2"]
+                                    ["val3" 4]]
+                             :separator ","
+                             :enclose-strings-in-quotes true)
+           "\"col1\",\"col2\"\n\"val1\",\"val2\"\n\"val3\",4\n")))
+
+  (testing "does not enclose nil values in quotes"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["val1" "val2"]
+                                    [nil "val4"]]
+                             :separator ","
+                             :enclose-strings-in-quotes true)
+           "\"col1\",\"col2\"\n\"val1\",\"val2\"\n,\"val4\"\n")))
+  
+  (testing "escapes quotes inside strings when enclosed in quotes"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["\"so called\" value" "val2"]
+                                    ["val3" "val4"]]
+                             :separator ","
+                             :enclose-strings-in-quotes true)
+           "\"col1\",\"col2\"\n\"\\\"so called\\\" value\",\"val2\"\n\"val3\",\"val4\"\n")))
+
+  (testing "does not escape quotes inside strings when not enclosed in quotes"
+    (is (= (csv/print-to-csv :column-names ["col1" "col2"]
+                             :rows [["\"so called\" value" "val2"]
+                                    ["val3" "val4"]]
+                             :separator ",")
+           "col1,col2\n\"so called\" value,val2\nval3,val4\n"))))
+
+(def ^:private applicant "applicant")
+(def ^:private test-time (DateTime. 1000))
+
+(deftest test-applications-to-csv
```

Comment:
Btw, this could be a non-integration non-db test if you created the application from events. See something like `rems.application.test-commands/test-submit` for example.

Reply:
However, a test that queried the API for applications in the db would be valuable.

Reply:
Added a test for API. I did not end up changing the existing test to non-integration, non-db right now, but added a TODO comment about it.